### PR TITLE
Tekton

### PR DESCRIPTION
**Pull request by csantanapr**
Here is the tekton pipeline which is equivalent to the Jenkinsfile

Here is how it looks in the tekton dashboard
image

### CODE REVIEW

**Comment by csantanapr** [comment]
Having one task with 11 steps, creates a single Pod with 12 containers. 


I'm thinking now that is proven to work, this can be refactor into multiple tasks, this can bring multiple benefits:
- smaller Pod, less resource requirements, Pod has better chance to run on any node with space to fit a small Pod
- re-use tasks in other pipelines (DRY), avoid duplication
- easier to troubleshoot by easily identifying a set of steps as a task and run in isolation with pipeline with a single task.

**Comment by csantanapr** [comment]
The tasks can be centralize in a central repo like the https://github.com/ibm-garage-cloud/ibm-garage-iteration-zero and only have the Pipeline definition in the repective template app repo like this repo.

**Comment by csantanapr** [comment]
added the input for `docker-image` as it is required by the tekton webhook/dashboard even it's not used.

for docker registry any value can be entered, it will be ignored.

And now I can push a change to git and triggers the pipeline

image

**Comment by csantanapr** [comment]
I refactored the pipeline, now is composed of reusable tasks/pods much smaller.
image


This Pipeline depends on the tasks from this other PR https://github.com/ibm-garage-cloud/ibm-garage-iteration-zero/pull/2

File: README.md [hunk 1]
```diff
@@ -63,6 +63,21 @@ Use the IBM Garage for Cloud CLI to register the GIT Repo with Jenkins environme
...
 ```
 
+### Tekton
+
+Create pipeline for each environment for example the `dev` namespace
+```bash
+kubectl create -f tekton/ -n dev
+```
+
+You need to install the taks that this pipeline depends from the iteration-zero repository:
```
Comment (mjperrins) [inline README.md]: taks spelt wrong

**Comment by mjperrins** [comment]
for next PR can we do the PR against the `dev` branch

**Pull request merged**